src/charm.py

```diff
@@ -431,7 +433,12 @@ def _agent_version(self) -> Optional[str]:
         if not self._container.can_connect():
             return None
         version_output, _ = self._container.exec(["/bin/agent", "-version"]).wait_output()
-        return re.search(r"v(\d*\.\d*\.\d*)", version_output).group(1)  # type: ignore
+        # Output looks like this:
+        # agent, version v0.26.1 (branch: HEAD, revision: 2b88be37)
+        result = re.search(r"v(\d*\.\d*\.\d*)", version_output)
```

Comment:
Is there a reason why we're stripping off the `v` at the beginning? It probably doesn't make a semantic difference, but we're a little hemmed in here.

Not that I ever really expect this to fail, but we're also checking to see if the container exec fails.

Either way, `v..` actually passes this regexp. I always personally prefer `groupdict` for legibility, even if it adds verbosity. Raw strings also great for regexps.

`re.search(r`version\s+(?P<version>.*?)\s+`).groupdict().get("version", None)` should do the same thing as a oneliner which is more or less immune to changes and grabs whatever whitespace-delimited string comes after "version", or `None` if nothing was matched.

Reply:
I find it a bit more simple as is so I will leave it for now. If the regex ever gets more complicated, we can change it as necessary.